packages/analytics-browser/src/storage/local-storage.ts

```diff
@@ -1,8 +1,26 @@
 import { getGlobalScope } from '@amplitude/analytics-client-common';
 import { BrowserStorage } from './browser-storage';
+import { Logger } from '@amplitude/analytics-types';
+import { MAX_ARRAY_LENGTH } from '../constants';
 
+interface LocalStorageOptions {
+  loggerProvider?: Logger;
+}
 export class LocalStorage<T> extends BrowserStorage<T> {
-  constructor() {
+  loggerProvider?: Logger;
+
+  constructor(config?: LocalStorageOptions) {
     super(getGlobalScope()?.localStorage);
+    this.loggerProvider = config?.loggerProvider;
+  }
+
+  async set(key: string, value: T): Promise<void> {
+    if (Array.isArray(value) && value.length > MAX_ARRAY_LENGTH) {
+      const droppedEventsCount = value.length - MAX_ARRAY_LENGTH;
+      await super.set(key, value.slice(droppedEventsCount) as T);
```

Comment:
Should it be

```suggestion
      await super.set(key, value.slice(0, droppedEventsCount) as T);
```

Reply:
Should we drop the most early received events like what we did in [JS SDK](https://github.com/amplitude/Amplitude-JavaScript/blob/4cbe557a81ca981d03e140bebed6134c49595a5e/src/amplitude-client.js#L1512-L1525)?

Reply:
sorry i mean to  say

```suggestion
      await super.set(key, value.slice(0, 1000) as T);
```

this should keep the top 1000 of the list

Reply:
```
await super.set(key, value.slice(droppedEventsCount) as T);
```
Feel like we should keep this and drop oldest events. For example:
[1, 2, 3, 4, 5, 6], with storage size 5, we drop the first element and save [2, 3, 4, 5, 6]

However, this drops the youngest events and keep [1,2,3,4,5] 
```
await super.set(key, value.slice(0, STORAGE_SIZE) as T);
```

Reply:
Reasons to use `slice(0, MAX_ARRAY_LENGTH)`
1. the event queue doesn't guarantee order of events because of retry logic. retried events get pushed at the back of the queue
2. not persisting in storage doesn't drop events, the events are still in memory
3. the first x events is because they are the ones about to be sent out hence have priority

Change the log info too